public/samples/DataStreams/StreamsUpkeep.sol

```diff
@@ -110,65 +98,48 @@ contract StreamsUpkeep is ILogAutomation, StreamsLookupCompatibleInterface {
+    // function will be performed on-chain
+    function performUpkeep(bytes calldata performData) external {
+        // Decode incoming performData
         (bytes[] memory signedReports, bytes memory extraData) = abi.decode(
             performData,
             (bytes[], bytes)
         );
 
         bytes memory report = signedReports[0];
 
-        bytes memory bundledReport = bundleReport(report);
+        (, bytes memory reportData) = abi.decode(report, (bytes32[3], bytes));
 
-        BasicReport memory unverifiedReport = _getReportData(report);
+        // Billing
+        IFeeManager feeManager = IFeeManager(address(verifier.s_feeManager()));
+        IRewardManager rewardManager = IRewardManager(
+            address(feeManager.i_rewardManager())
+        );
 
-        bytes memory verifiedReportData = verifier.verify{
-            value: unverifiedReport.nativeFee
-        }(bundledReport);
-        BasicReport memory verifiedReport = abi.decode(
-            verifiedReportData,
-            (BasicReport)
+        address feeTokenAddress = feeManager.i_linkAddress();
+        (Common.Asset memory fee, , ) = feeManager.getFeeAndReward(
+            address(this),
+            reportData,
+            feeTokenAddress
         );
 
-        emit PriceUpdate(verifiedReport.price);
-    }
+        // Approve fee spend
+        IERC20(feeTokenAddress).approve(address(rewardManager), fee.amount);
 
-    function bundleReport(
-        bytes memory report
-    ) internal view returns (bytes memory) {
-        Quote memory quote;
-        quote.quoteAddress = FEE_ADDRESS;
-        (
-            bytes32[3] memory reportContext,
-            bytes memory reportData,
-            bytes32[] memory rs,
-            bytes32[] memory ss,
-            bytes32 raw
-        ) = abi.decode(
-                report,
-                (bytes32[3], bytes, bytes32[], bytes32[], bytes32)
-            );
-        bytes memory bundledReport = abi.encode(
-            reportContext,
-            reportData,
-            rs,
-            ss,
-            raw,
-            abi.encode(quote)
+        // Verify the report
+        bytes memory verifiedReportData = verifier.verify(
+            report,
+            abi.encode(feeTokenAddress)
         );
-        return bundledReport;
-    }
 
-    function _getReportData(
-        bytes memory signedReport
-    ) internal pure returns (BasicReport memory) {
-        (, bytes memory reportData, , , ) = abi.decode(
-            signedReport,
-            (bytes32[3], bytes, bytes32[], bytes32[], bytes32)
+        // Decode verified report data into BasicReport struct
+        BasicReport memory verifiedReport = abi.decode(
+            verifiedReportData,
+            (BasicReport)
         );
 
-        BasicReport memory report = abi.decode(reportData, (BasicReport));
-        return report;
+        // Log price from report
+        emit PriceUpdate(verifiedReport.price);
```

Comment:
if you accept my suggestion on also storing as a storage var, then assign here.

Reply:
Kept both the log and also added the storage variable